keras_nlp/models/gpt2/gpt2_preprocessing_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for GPT-2 preprocessing layers."""
+
+import os
+
+import tensorflow as tf
+from absl.testing import parameterized
+from tensorflow import keras
+
+from keras_nlp.models.gpt2.gpt2_preprocessing import GPT2Tokenizer
+
+
+class GPT2TokenizerTest(tf.test.TestCase, parameterized.TestCase):
+    def setUp(self):
+        vocab = {
+            "<|endoftext|>": 0,
+            "Ġair": 1,
+            "plane": 2,
+            "Ġat": 3,
+            "port": 4,
+            "Ġkoh": 5,
+            "li": 6,
+            "Ġis": 7,
+            "Ġthe": 8,
+            "Ġbest": 9,
+        }
+
+        merges = ["Ġ a", "Ġ t", "Ġ k", "Ġ i", "Ġ b", "Ġa i", "p l", "n e"]
+        merges += ["Ġa t", "p o", "r t", "o h", "l i", "Ġi s", "Ġb e", "s t"]
+        merges += ["Ġt h", "Ġai r", "pl a", "Ġk oh", "Ġth e", "Ġbe st", "po rt"]
+        merges += ["pla ne"]
+
+        self.tokenizer = GPT2Tokenizer(vocabulary=vocab, merges=merges)
+
+    def test_tokenize(self):
+        input_data = " airplane at airport"
+        output = self.tokenizer(input_data)
+        self.assertAllEqual(output, [1, 2, 3, 1, 4])
+
+    def test_tokenize_batch(self):
+        input_data = tf.constant([" airplane at airport", " kohli is the best"])
+        output = self.tokenizer(input_data)
+        self.assertAllEqual(output, [[1, 2, 3, 1, 4], [5, 6, 7, 8, 9]])
+
+    def test_detokenize(self):
+        input_tokens = [[1, 2, 3, 1, 4]]
```

Comment:
I know this is probably prexisting from the BERT texts I wrote :p, but would `input_tokens = [1, 2, 3, 1, 4]` and `self.assertEqual(output, " airplane at airport")` work? That would make this test the actual inverse of `test_tokenize`.